src/curl_util.cpp

```diff
@@ -122,6 +122,7 @@ struct curl_slist* curl_slist_remove(struct curl_slist* list, const char* key)
             struct curl_slist *tmp = *p;
             *p = (*p)->next;
```

Comment:
Does this segfault because the pointer is advanced twice, one here and once on line 112?  So this logic is wrong?

Also is it possible to handle the situation with multiple list nodes with the name key?

Could you also add a unit test?

Reply:
@gaul I commented, but this seems to be due to a simple bug.
First of all, I think it's a good idea to have my suggested code tried.

Reply:
@gaul @ggtakec 
Thanks for your comments.
This is a bug in the code, after removing the key from the `requestHeaders` the pointer `*p` is getting advanced twice. 
Currently my code change is breaking the loop assuming there is always single instance of a key in the nodes of `requestHeaders` list.